api4/cloud.go

```diff
@@ -475,6 +475,12 @@ func handleCWSWebhook(c *Context, w http.ResponseWriter, r *http.Request) {
 			c.Err = appErr
 			return
 		}
+	case model.EventTypeProductLimitsChanged:
```

Comment:
Tested through test in enterprise.

Reply:
Maybe? I see 3 options.

1) We could update the sub here now. It seems safe now while limits and subs are joined at the hip, but becomes brittle in the future if we do things like condition limits upon special one-off things we apply at a customer level (like incentivized trials). If we are setting ourselves up to support that now, I'd wan't to add more fields to the event like a nillable sub and the reason the limits changed.

2) If we leave this as implemented now, the subscription cache in mm-server is very short-lived, 5 minutes. In that window, an admin would see out of date info in the system console, but I think most other UI would be ok.

3) As an easy compromise could be to always evict the subscription cache without renewing it. Then the only time inconsistent data would be seen is if the admin is already in the system console at the time the subscription changes, and only until they refresh the page.

Reply:
Will review these once we have the CWS one again so I can get a complete picture

Reply:
Tried adding the sub to the event and updating it here, too. Since all properties are nillable on the event, looks like it is as easy as updating the sub in the cache if its in the event, and leave it alone otherwise. Right now, it is always present because the only webhook updates we're sending are because the subscription change made it change. Opened the CWS PR and updated these 3 PRs to update the subscription.